lib/platform/linux-i2c.c

```diff
@@ -373,18 +374,29 @@ static void i2c_gas_write(struct switchtec_dev *dev, void __gas *dest,
 	struct switchtec_i2c *idev = to_switchtec_i2c(dev);
 	uint8_t tag;
 	uint8_t status;
-	uint8_t retry_count = 0;
+	uint8_t write_retry_count = 0;
+	uint8_t status_retry_count = 0;
 
 	do {
+		status_retry_count = 0;
 		tag = get_tag(idev);
 		i2c_gas_data_write(dev, dest, src, n, tag);
-		status = i2c_gas_write_status_get(dev, tag);
-		if (status == 0 || status == GAS_TWI_MRPC_ERR)
+		do {
+			usleep(5000);
+			status = i2c_gas_write_status_get(dev, tag);
+			if (status == 0 || status == GAS_TWI_MRPC_ERR)
+				break;
+			status_retry_count++;
+		} while (status_retry_count < MAX_STATUS_GET_RETRY);
```

Comment:
The nested retry loop seems nasty.... At the very *least* push it into it's own function.

Reply:
Agree.